[[ Gypify ]] Gypify missing lcidlc project file

montegoulding opened the pull request:
The lcidlc project files had been removed from the repo but not replaced
with gyp files.

Commit:
[[ Gypify ]] Gypify missing lcidlc project file

The lcidlc project files had been removed from the repo but not replaced
with gyp files.

runrevmark commented:
@montegoulding: Thanks for this :)

runrevmark commented:
@livecode-vulcan review ok cfdf422

peter-b commented:
Sorry, this breaks the Linux build.

@livecode-vulcan review not ok cfdf422

runrevmark commented:
@montegoulding: As Peter points out - Linux isn't building here. Looking at the errors, I think its because the derived files aren't being placed / processed in the right place. As EncodedSupport.c etc. are not part of the source tree, they shouldn't be in the 'src' directly on processing. Perhaps @livecodefraser can advise?

livecode-vulcan commented:
💔 review by @peter-b not ok cfdf422b667f68935722797393cbc5028602b928

livecodefraser commented:
Hi Monte,

Intermediate sources should get placed in a directory inside <(SHARED_INTERMEDIATE_DIR), e.g. '<(SHARED_INTERMEDIATE_DIR)/LCIDLC/EncodedSupport.c'. You'll need to update both the outputs from the actions and the list of sources that includes these.

(It depends on the build system, but you should use <(SHARED_INTERMEDIATE_DIR) instead of <(INTERMEDIATE_DIR) as the non-shared directory is only guaranteed to be shared within a target.)

montegoulding commented:
Ok thanks

Commit:
[[ Gypify ]] Generate files into intermediate directory

livecodefraser commented:
@livecode-vulcan review okay 7d19b96

Thanks!

livecode-vulcan commented:
💙 review by @livecodefraser ok 7d19b9602eabc070d518ca4fbacf2e604a7331cc

livecode-vulcan commented:
😞 test failure 7d19b9602eabc070d518ca4fbacf2e604a7331cc
- try-community-linux-x86: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/676)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/689)
- try-community-linux-x86_64: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/677)
- try-community-emscripten: [failure](https://vulcan.livecode.com/builders/try-community-emscripten/builds/679)
- try-community-ios: [failure](https://vulcan.livecode.com/builders/try-community-ios/builds/689)
- try-community-win-x86: [pending](https://vulcan.livecode.com/builders/try-community-win-x86/builds/700)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/691)

montegoulding commented:
Hmm... obviously this doesn't need to build for iOS or emscripten. Should I make it a dependency of the development build or do I need to do something funky with killing targets unless they are mac, linux or win?

peter-b committed:
Don't build lcidlc on emscripten or iOS platforms

`lcidlc` isn't needed for either of these operating systems.

Commit:
Merge pull request #1 from peter-b/patch-1

Don't build lcidlc on emscripten or iOS platforms

Commit:
Don't compile lcidlc for android

runrevmark commented on lcidlc/lcidlc.gyp:
```diff
@@ -0,0 +1,125 @@
+				'src/Value.h',
+				'src/Value.cpp',
+
+                '<(SHARED_INTERMEDIATE_DIR)/LCIDLC/EncodedJavaSupport.c',
```
I'd make sure the folder is lower-case - i.e. lcidlc - just to make sure things are consistent on case-sensitive filesystems.

montegoulding replied:
Changed

livecodefraser commented on lcidlc/lcidlc.gyp:
```diff
@@ -0,0 +1,125 @@
+					'OS == "android" or OS == "ios" or OS == "emscripten"',
+					{
+						'type': 'none',
+					},
```
Looks like there's a missing ']' here (JSON definitely isn't the easiest build configuration syntax to read!).

montegoulding replied:
Should be fixed now. Would be handy if it were standard JSON so we could run a validator on it. It seemed to work so obviously gyp isn't overly fussy...

Commit:
Fix formatting issues and lowercase intermediate dir

livecodefraser commented:
@livecode-vulcan review okay 2f64e97

livecode-vulcan commented:
💙 review by @livecodefraser ok 2f64e97359588150d24e3a0ba0d1cd80142257ec

livecode-vulcan commented:
😞 test failure 2f64e97359588150d24e3a0ba0d1cd80142257ec
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/683)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/696)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/684)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/686)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/696)
- try-community-win-x86: [failure](https://vulcan.livecode.com/builders/try-community-win-x86/builds/707)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/698)

montegoulding commented:
Hmm... Is there any way to get any more info about the failure? The failure link goes to a blank page.

livecodefraser commented:
It looks like the output EncodedJavaSupport.c and EncodedSupport.c have some some syntax that the MSVC compiler is objecting to. I'll have a look at what the encoding step is producing to see what is wrong.

montegoulding commented:
Ah, it may not have been built on anything but mac for a good long while. Perhaps before moving to perl for the encoding of those files...

livecodefraser commented:
const char *......\lcidlc\g_java_support_template [] =
{ ... };

I can see why MCVC is unhappy ;)

livecodefraser commented:
Ah, looks like one of those really annoying Gyp-isms... the third parameter to the script is being treated as a file path and is having the relative path prefixed to it. I can't remember what I did to fix this last time I encountered this so I need to do a bit of spelunking.

montegoulding commented:
ah... and only on win it seems

montegoulding commented:
would wrapping in double quotes help?

livecodefraser committed:
[[ Gyp-LCIDLC ]] Work around a Gyp bug in passing command arguments

On Windows, Gyp is sometimes too eager in interpreting arguments
to scripts as paths rather than literals. As I've not been able
to track down the source of this issue, a work-around in the
affected script is required.

livecodefraser committed:
[[ Gyp-LCIDLC ]] Windows: ensure LCIDLC is build as a console application

livecodefraser commented:
There is a bug in Gyp on Windows where it is sometimes too eager in relativising paths (and I've not yet been able to track down why :( ).

A tweak is also needed to make sure that LCIDLC is built as a console-subsystem app instead of a Win32 one, otherwise it fails to link because WinMain is missing.

The first has been worked around and the second fixed in the PR I've submitted against your branch.

Commit:
Merge pull request #2 from livecodefraser/feature/gypify_lcidl-win_fixes

Fixes for Windows-specific compilation issues

montegoulding commented:
Great, thanks.

Commit:
Add comment about workaround

livecodefraser commented:
Isn't Gyp fun?

@livecode-vulcan review okay 64ff620

livecode-vulcan commented:
💙 review by @livecodefraser ok 64ff6209ed0746b39546cd0ceaaa1c933bc4c510

montegoulding commented:
@livecodefraser It looked like fun for a little while ;-)

montegoulding commented:
BTW is there any way contributors can test a branch against vulcan before submitting a PR?

peter-b commented:
@montegoulding Containerisation on Macs sucks, so unfortunately our mac builds boil down to "run this arbitrary code as a normal user account".  At the moment we're only building reviewed code on our infrastructure.  Sorry... :disappointed: You can test using Travis CI for now, but when I have time (hah!) I really need to set things up so that unprivileged users can request test builds on the "safe" targets.

livecodefraser commented:
Peter pre-empted my response there! Only thing I'll add is that I need to get around to writing up the instructions for setting up a Windows build environment (which is only useful if you have a copy of Windows available to you, but it's a start at least).

peter-b commented:
@livecodefraser Out of interest, are all the bits needed for a Windows build free downloads now, or do you need a MSDN subscription?

livecodefraser commented:
Saying that... if you're feeling brave and can get a copy of Visual Studio 2010 (we use the Professional edition but the free Express might work...), you can do as we do on Vulcan and build on Linux under Wine!  Or even Wine for OSX, but that might be going a bit too far.

livecode-vulcan commented:
😎 test success 64ff6209ed0746b39546cd0ceaaa1c933bc4c510
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/686)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/699)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/687)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/689)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/699)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/710)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/701)

montegoulding commented:
OK. No problem re-vulcan. I usually work on Mac so that's the one I don't really need. Having a way to work out that I need to jump on another machine to sort something out would be handy... 

I've enabled travis now

Re windows... have things changed since gypification? Last time I built on Win I used VCE 2005.

livecodefraser commented:
Yes, we updated to VS2010 as part of Gypification. Partly because Gyp has better support for the VS2010 project file format and partly because VS2005 was showing its age. We haven't gone any newer yet as 2010 was the last version that supports running on XP.
